core/src/upgrade/apply.rs

```diff
@@ -74,7 +112,7 @@ where
         .protocol_info()
         .into_iter()
         .map(NameWrap as fn(_) -> NameWrap<_>);
-    let future = multistream_select::dialer_select_proto(conn, iter, v);
+    let future = multistream_select::dialer_select_proto(conn, iter, v.into());
```

Comment:
I was considering suggesting to create a dedicated `dialer_sim_open_select_proto` function but I guess that idea is not particularly good because we rely on being able to name all these futures here?